twilight-lavalink/src/model/incoming.rs

```diff
@@ -0,0 +1,371 @@
+    /// An update about the information of a player.
+    PlayerUpdate(PlayerUpdate),
+}
+
+impl From<Ready> for IncomingEvent {
+    fn from(event: Ready) -> IncomingEvent {
+        Self::Ready(event)
+    }
+}
+
+impl From<Event> for IncomingEvent {
+    fn from(event: Event) -> IncomingEvent {
+        Self::Event(event)
+    }
+}
+
+impl From<PlayerUpdate> for IncomingEvent {
+    fn from(event: PlayerUpdate) -> IncomingEvent {
+        Self::PlayerUpdate(event)
+    }
+}
+
+impl From<Stats> for IncomingEvent {
+    fn from(event: Stats) -> IncomingEvent {
+        Self::Stats(event)
+    }
+}
+
+/// The discord voice information that lavalink uses for connection and sending
+/// information.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct VoiceState {
+    /// The Discord voice endpoint to connect to.
+    pub endpoint: String,
+    /// The Discord voice session id to authenticate with. Note this is separate
+    /// from the lavalink session id.
+    pub session_id: String,
+    /// The Discord voice token to authenticate with.
+    pub token: String,
+}
+
+/// An update of a player's status and state.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct PlayerUpdate {
+    /// The guild ID of the player.
+    pub guild_id: Id<GuildMarker>,
+    /// The new state of the player.
+    pub state: PlayerUpdateState,
+}
+
+impl PlayerUpdate {
+    /// The operation type of the `PlayerUpate` event.
+    pub const OPCODE: Opcode = Opcode::PlayerUpdate;
+}
+
+/// New statistics about a node and its host.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct PlayerUpdateState {
+    /// True when the player is connected to the voice gateway.
+    pub connected: bool,
+    /// The ping of the node to the Discord voice server in milliseconds (-1 if not connected).
+    pub ping: i64,
+    /// Track position in milliseconds. None if not playing anything.
+    pub position: i64,
+    /// Unix timestamp of the player in milliseconds.
+    pub time: i64,
+}
+
+/// Dispatched by Lavalink upon successful connection and authorization.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct Ready {
+    /// Whether this session was resumed.
+    pub resumed: bool,
+    /// The Lavalink session id of this connection. Not to be confused with a
+    /// Discord voice session id.
+    pub session_id: String,
+}
+
+/// Statistics about a node and its host.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct Stats {
+    /// CPU information about the node's host.
+    pub cpu: StatsCpu,
+    /// Statistics about audio frames.
+    #[serde(skip_serializing_if = "Option::is_none")]
```

Comment:
are we ever sending this payload? If not, this feels kinda useless and so do the serialize derives on all these